File: api/src/main/java/com/coffeeshop/api/service/AllergenService.java
package com.coffeeshop.api.service;

import com.coffeeshop.api.model.AllergenEntity;
import com.coffeeshop.api.repository.AllergenRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class AllergenService {

    private final AllergenRepository allergenRepository;

    public List<AllergenEntity> getMilkAllergens(Long milkId) {
        return allergenRepository.getMilkAllergens(Math.toIntExact(milkId));
    }

    public List<AllergenEntity> getSyrupAllergens(Long syrupId) {
        return allergenRepository.getSyrupAllergens(Math.toIntExact(syrupId));
    }

    public List<AllergenEntity> getFillingAllergens(Long fillingId) {
        return allergenRepository.getFillingAllergens(Math.toIntExact(fillingId));
    }

}

File: front/src/app/services/allergen.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {AllergenModel} from "../model/AllergenModel";

@Injectable({
  providedIn: 'root'
})
export class AllergenService {

  url = 'http://localhost:52520/api/allergen';

  constructor(private http: HttpClient) {
  }

  getSyrupAllergens(syrupId: number) {
    return this.http.get<AllergenModel[]>(this.url + `/syrup?syrup_id=${syrupId}`,{
      withCredentials: true
    })
  }

  getMilkAllergens(milkId: number) {
    return this.http.get<AllergenModel[]>(this.url + `/milk?milk_id=${milkId}`,{
      withCredentials: true
    })
  }

  getFillingAllergens(fillingId: number) {
    return this.http.get<AllergenModel[]>(this.url + `/filling?filling_id=${fillingId}`,{
      withCredentials: true
    })
  }

}
